examples/tv-casting-app/tv-casting-common/core/CastingApp.cpp

```diff
@@ -0,0 +1,119 @@
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "CastingApp.h"
+
+#include <app/clusters/bindings/BindingManager.h>
+#include <app/server/Server.h>
+#include <credentials/DeviceAttestationCredsProvider.h>
+#include <credentials/attestation_verifier/DeviceAttestationVerifier.h>
+
+namespace matter {
+namespace casting {
+namespace app {
+
+using namespace matter::casting::support;
+
+CastingApp * CastingApp::castingApp_ = nullptr;
+
+CastingApp::CastingApp() {}
+
+CastingApp * CastingApp::GetInstance()
+{
+    if (castingApp_ == nullptr)
+    {
+        castingApp_ = new CastingApp();
+    }
+    return castingApp_;
+}
+
+CHIP_ERROR CastingApp::Initialize(const AppParameters & appParameters)
+{
+    VerifyOrReturnError(mState == UNINITIALIZED, CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrReturnError(appParameters.GetCommissionableDataProvider() != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrReturnError(appParameters.GetDeviceAttestationCredentialsProvider() != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrReturnError(appParameters.GetServerInitParamsProvider() != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+
+    mAppParameters = &appParameters;
+
+    ReturnErrorOnFailure(chip::Platform::MemoryInit());
+    ReturnErrorOnFailure(chip::DeviceLayer::PlatformMgr().InitChipStack());
+
+    chip::DeviceLayer::SetCommissionableDataProvider(appParameters.GetCommissionableDataProvider());
+    chip::Credentials::SetDeviceAttestationCredentialsProvider(appParameters.GetDeviceAttestationCredentialsProvider());
+    chip::Credentials::SetDeviceAttestationVerifier(appParameters.GetDeviceAttestationVerifier());
+
+#if CHIP_ENABLE_ROTATING_DEVICE_ID
+    MutableByteSpanDataProvider * uniqueIdProvider = appParameters.GetRotatingDeviceIdUniqueIdProvider();
+    if (uniqueIdProvider != nullptr && uniqueIdProvider->Get() != nullptr)
+    {
+        ReturnErrorOnFailure(chip::DeviceLayer::ConfigurationMgr().SetRotatingDeviceIdUniqueId(*uniqueIdProvider->Get()));
+    }
+#endif // CHIP_ENABLE_ROTATING_DEVICE_ID
+
+    mState = NOT_RUNNING; // initialization done, set state to NOT_RUNNING
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR CastingApp::Start()
+{
+    VerifyOrReturnError(mState == NOT_RUNNING, CHIP_ERROR_INCORRECT_STATE);
+    auto & server = chip::Server::GetInstance();
+
+    // start Matter server
+    chip::ServerInitParams * serverInitParams = mAppParameters->GetServerInitParamsProvider()->Get();
+    VerifyOrReturnError(serverInitParams != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+    ReturnErrorOnFailure(server.Init(*serverInitParams));
+
+    // TBD: Set CastingApp as AppDelegate
+    // &server.GetCommissioningWindowManager().SetAppDelegate(??);
```

Comment:
Instead of TBD, shouldn't these be marked as TODOs?

TBD sounds like there's a decision to make

Reply:
Done.